wallet/db.c

```diff
@@ -359,6 +359,9 @@ char *dbmigrations[] = {
     /* Fix dangling peers with no channels. */
     "DELETE FROM peers WHERE id NOT IN (SELECT peer_id FROM channels);",
     "ALTER TABLE outputs ADD scriptpubkey BLOB;",
+    /* #2316 feerate per channel */
+    "ALTER TABLE channels ADD feerate_base INTEGER;",
+    "ALTER TABLE channels ADD feerate_ppm INTEGER;",
```

Comment:
Yeah, this lacks sophistication.  We really want to set these to the current global values, but our db upgrade code is dumb.  @cdecker might have ideas, but perhaps each upgrade should be a pair: a string and a function, either of which can be NULL.  The function in this case would initialize those feerate_ vars correctly?

Reply:
hmmm..  I see what you are up to. "DB: more advanced upgrade" #2394  would allow me to read out the current global values and set all channels specific feerate values to the current global on DB update.

I kinda dislike the idea, because if on a existing clightning node someone changes the gobal feerate config, it will not affect **any** of exiting channels, which is probably not what the user expects here. Is this really what you want?

My suggestion is maybe to have a third DB value indicating the usage of specific or global fees or maybe allow -1 for that matter. This way specific rates will only apply when set and not disabled yet, global values are normal operation 'fallback'.

Reply:
My suggestion since 2394 is not merged yet and we can have a global fallback anytime:

https://github.com/ElementsProject/lightning/pull/2342/commits/40af7d1910d892d2b7d40349fd52b8efa463c00c

Reply:
No, I'm happy for the change in sematics: now we have per-channel control, the global simply becomes a default-at-creation.  It's unusual for people to modify them at the moment anyway.

Reply:
OK, thanks for the clarification